Wikipedia/Code/InsertMediaSearchResultsCollectionViewController.swift

```diff
@@ -0,0 +1,230 @@
+        title = CommonStrings.searchTitle
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        collectionView.register(InsertMediaSearchResultCollectionViewCell.self, forCellWithReuseIdentifier: InsertMediaSearchResultCollectionViewCell.identifier)
+        view.wmf_addSubviewWithConstraintsToEdges(collectionView)
+        additionalSafeAreaInsets = UIEdgeInsets(top: 0, left: 0, bottom: flowLayout.minimumLineSpacing, right: 0)
+        collectionView.contentInset = UIEdgeInsets(top: 0, left: 12, bottom: 0, right: 12)
+        apply(theme: theme)
+    }
+
+    func reload() {
+        collectionView.reloadData()
+        updateEmptyState()
+    }
+
+    private func configure(_ cell: InsertMediaSearchResultCollectionViewCell, at indexPath: IndexPath) {
+        let result = searchResults[indexPath.item]
+        cell.configure(imageURL: result.thumbnailURL, imageViewDimension: flowLayout.itemSize.width, caption: result.displayTitle)
+        cell.apply(theme: theme)
+    }
+
+    func setImageInfo(_ imageInfo: MWKImageInfo?, for searchResult: InsertMediaSearchResult, at index: Int) {
+        assert(Thread.isMainThread)
+        searchResult.imageInfo = imageInfo
+    }
+
+    // MARK: Themeable
+
+    override func apply(theme: Theme) {
+        super.apply(theme: theme)
+        view.backgroundColor = theme.colors.paperBackground
+        collectionView.backgroundColor = theme.colors.paperBackground
+    }
+
+    // MARK: - Empty State
+
+    var emptyViewType: WMFEmptyViewType = .noSearchResults
+
+    final var isEmpty = true
+    final var showingEmptyViewType: WMFEmptyViewType?
+    final func updateEmptyState() {
+        let sectionCount = numberOfSections(in: collectionView)
+
+        var isCurrentlyEmpty = true
+        for sectionIndex in 0..<sectionCount {
+            if self.collectionView(collectionView, numberOfItemsInSection: sectionIndex) > 0 {
+                isCurrentlyEmpty = false
+                break
+            }
+        }
+
+        guard isCurrentlyEmpty != isEmpty || showingEmptyViewType != emptyViewType else {
+            return
+        }
+
+        isEmpty = isCurrentlyEmpty
+
+        isEmptyDidChange()
+    }
+
+    private var emptyViewFrame: CGRect {
+        let insets = scrollView?.contentInset ?? UIEdgeInsets.zero
+        let frame = view.bounds.inset(by: insets)
+        return frame
+    }
+
+    open func isEmptyDidChange() {
+        if isEmpty {
+            wmf_showEmptyView(of: emptyViewType, action: nil, theme: theme, frame: emptyViewFrame)
+            showingEmptyViewType = emptyViewType
+        } else {
+            wmf_hideEmptyView()
+            showingEmptyViewType = nil
+        }
+    }
+
+    override func scrollViewInsetsDidChange() {
+        super.scrollViewInsetsDidChange()
+        wmf_setEmptyViewFrame(emptyViewFrame)
```

Comment:
Once the insets are taken into account in the flow layout, there would need to be a flow layout invalidation here either using a context or updating the flow layout subclass to handle the context-less method 
```
let context = UICollectionViewFlowLayoutInvalidationContext()
context.invalidateFlowLayoutAttributes = true
flowLayout.invalidateLayout(with: context)
```